src/doc/rustc/src/platform-support/x86_64-pc-cygwin.md

```diff
@@ -0,0 +1,44 @@
+# `x86_64-pc-cygwin`
+
+**Tier: 3**
+
+Windows targets supporting Cygwin and MSYS2.
+The `*-cygwin-*` targets are **not** intended as native targets for applications,
+a developer writing Windows applications should use the `*-pc-windows-*` targets instead, which are *native* Windows.
+
+Cygwin is only intended as an emulation layer for UNIX-only programs which do not support the native Windows targets.
+
+## Target maintainers
+
+- [Berrysoft](https://github.com/Berrysoft)
+
+## Requirements
+
+This target is cross compiled. Different `target_env` needs different linker:
+- `x86_64-pc-cygwin`: `x86_64-pc-cygwin-gcc`
+- `x86_64-pc-cygwin-msys2`: `x86_64-pc-msys-gcc`
+
+The difference between two targets are small.
```

Comment:
Could you mention what the actual difference is here, or how to select between the two? It may be good to mention how this compares to the somewhat confusable `pc-windows-gnu` target as well.